From 4c81f8af352f6e3f1040371f8304914e394022a5 Mon Sep 17 00:00:00 2001
From: Gu, Wangyi <wangyi.gu@intel.com>
Date: Tue, 15 Jan 2013 10:01:22 +0800
Subject: [PORT FROM MAIN] Movie studio: fix a memory leak issue in the AOSP code

BZ: 80150

It's an android defect, connect()/disconnect() of OMXClient need be called in a pair,
if miss calling disconnect() while connect() is called, a memory leak will happen.

Change-Id: I9acc19242faa5c981bb77caba52170e0f96dbf96
Orig-Change-Id: I0fa40422aa54dc97dadd850e8a4cb5a9697cb685
Signed-off-by: Gu, Wangyi <wangyi.gu@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 .../src/VideoEditorVideoDecoder.cpp                |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp b/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp
index de91731..1845480 100755
--- a/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp
+++ b/libvideoeditor/vss/stagefrightshells/src/VideoEditorVideoDecoder.cpp
@@ -779,6 +779,7 @@ M4OSA_ERR queryVideoDecoderCapabilities
     logSupportDecodersAndCapabilities(pDecoders);
     *decoders = pDecoders;
 cleanUp:
+    client.disconnect();
     return err;
 }
 /********************
-- 
1.7.4.1

